lego/apps/polls/models.py

```diff
@@ -0,0 +1,65 @@
+from django.conf import settings
+from django.db import models, transaction
+from django.db.models import Sum
+from django.utils import timezone
+from rest_framework.exceptions import ParseError
+
+from lego.apps.content.models import Content
+from lego.apps.polls.permissions import PollPermissionHandler
+from lego.apps.users.models import User
+from lego.utils.models import BasisModel
+
+
+def get_time_delta():
+    return timezone.now() + timezone.timedelta(weeks=52)
+
+
+class Poll(Content, BasisModel):
+
+    description = models.TextField(blank=True)
+
+    valid_until = models.DateTimeField(default=get_time_delta)
+
+    answered_users = models.ManyToManyField(User, related_name="answered_polls")
+
+    def __str__(self):
+        return self.title
+
+    @property
+    def total_votes(self):
+        return Option.objects.filter(poll=self).aggregate(total_votes=Sum("votes"))[
+            "total_votes"
+        ]
+
+    def get_has_answered(self, user):
+        return self.answered_users.all().filter(pk=user.id).exists()
+
+    def get_absolute_url(self):
+        return f"{settings.FRONTEND_URL}/polls/{self.id}/"
+
+    @transaction.atomic
+    def vote(self, user, option_id):
+        option = self.options.get(pk=option_id)
+        if not option:
+            raise ParseError(detail="Option not found.")
+        option.votes += 1
```

Comment:
The votes are increased from the model without checking if the user has voted already, should we allow that or should we move that logic in here too?

Reply:
It is checked in the view already, it is neccesary to verify it here as well?

Reply:
I just wondered if there was any reason to have one check in this method (the option exists) and rely on that the user is checked elsewhere. Is there any reason not to check both here (and not in the view)? But then again, might be an anti-pattern I'm not aware of.

Reply:
I guess it is cleaner and safer to have all validation in the model.